postgres/src/main/java/org/jdbi/v3/postgres/InetColumnMapper.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.postgres;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import org.jdbi.v3.core.mapper.ColumnMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.postgresql.util.PGobject;
+
+/**
+ * A column mapper which maps Postgres's {@code inet} type to Java's {@link InetAddress}.
+ */
+class InetColumnMapper implements ColumnMapper<InetAddress> {
+    @Override
+    public InetAddress map(ResultSet r, int columnNumber, StatementContext ctx) throws SQLException {
+        String val = Optional.ofNullable(r.getObject(columnNumber))
+            .map(PGobject.class::cast)
+            .map(PGobject::getValue).orElse(null);
+
+        try {
+            return val == null ? null : InetAddress.getByName(val);
```

Comment:
this makes me super-nervous. I am not sure that we want to do actual DNS resolution on the fly when mapping data from the database. It also only returns a single value, there is a method (getAllByName) that returns an array of names (which makes sense as these names can map to multiple addresses).

But then again, we already do that in the InternetMapperFactory, so I guess the point is moot.

Reply:
`InternetMapperFactory` does already what is needed. I'll use that (must make it public though).